mod_reforged/hooks/skills/perks/perk_anticipation.nut

```diff
@@ -0,0 +1,115 @@
+::mods_hookExactClass("skills/perks/perk_anticipation", function(o) {
+	// Public
+	o.m.UsesMax <- 2;
+	o.m.DamageReductionPerTile <- 10.0;
+	o.m.DamageReductionBase <- 0.0;
+
+	// Private
+	o.m.UsesRemaining <- 2;
+
+	// Temp
+	o.m.TempDamageReduction <- 0.0;
+	o.m.IsAboutToConsumeUse <- false;
+
+	local create = o.create;
+	o.create = function()
+	{
+		create();
+		this.m.IconMini = "rf_anticipation_mini";
+
+		this.addType(::Const.SkillType.StatusEffect);	// We now want this effect to show up on entities
+	}
+
+	o.isHidden <- function()
+	{
+		return (this.isEnabled() == false);
+	}
+
+	o.onCombatStarted <- function()
+	{
+		this.m.UsesRemaining = this.m.UsesMax;
+	}
+
+	o.onCombatFinished <- function()
+	{
+		this.m.UsesRemaining = this.m.UsesMax;	// So that for the purposes of the tooltip everything looks good
+	}
+
+	o.getDescription <- function()
+	{
+		return ("Take reduced damage from the first " + ::MSU.Text.colorGreen(this.m.UsesMax) + " attacks you receive each battle. This reduction is dependant on your current Ranged Defense and the distance between the attacker and you");
```

Comment:
Try to avoid overwriting `getDescription` as much as possible. Instead of this, set `this.m.Description` in the `create` hook to `Use your fast senses to anticipate attacks against you, taking reduced damage.` or something like that so that it is a static string.

Then in the `getTooltip` function, add an entry that shows how many next attacks the effect is valid for.